Marsiple/App/Todos/TodoTableViewCell.swift

```diff
@@ -36,29 +41,61 @@ extension TodoTableViewCell {
     }
 }
 
+private extension TodoTableViewCell {
+    @objc func titleTapped() {
+        delegate?.tableViewCell(didSelectTitleAt: indexPath)
+    }
+    
+    @objc func checkBoxTapped() {
+        delegate?.tableViewCell(didSelectCheckBoxAt: indexPath)
+    }
+}
+
 private extension TodoTableViewCell {
     func setupViews() {
+        setupContainerViews()
         setupTodoLabel()
         setupCheckBoxImageView()
     }
     
+    func setupContainerViews() {
+        contentView.addSubview(todoLabelContainerView)
+        todoLabelContainerView.snp.makeConstraints {
+            $0.leading.top.bottom.equalToSuperview()
+        }
+        contentView.addSubview(checkBoxContainerView)
+        checkBoxContainerView.setContentHuggingPriority(UILayoutPriority.required, for: .horizontal)
+        checkBoxContainerView.snp.makeConstraints {
+            $0.top.trailing.bottom.equalToSuperview()
+            $0.leading.equalTo(todoLabelContainerView.snp.trailing)
+        }
+    }
+    
     func setupTodoLabel() {
-        contentView.addSubview(todoLabel)
+        todoLabelContainerView.addSubview(todoLabel)
         todoLabel.snp.makeConstraints {
-            $0.leading.top.bottom.equalToSuperview().inset(10)
+            $0.edges.equalToSuperview().inset(10)
         }
     }
     
     func setupCheckBoxImageView() {
         checkBoxImageView.contentMode = .scaleAspectFit
         checkBoxImageView.clipsToBounds = true
         checkBoxImageView.image = #imageLiteral(resourceName: "checkbox-unselected")
-        contentView.addSubview(checkBoxImageView)
+        checkBoxContainerView.addSubview(checkBoxImageView)
         checkBoxImageView.setContentHuggingPriority(UILayoutPriority.required, for: .horizontal)
         checkBoxImageView.snp.makeConstraints {
-            $0.leading.equalTo(todoLabel.snp.trailing).inset(-10)
-            $0.top.trailing.bottom.equalToSuperview().inset(10)
+            $0.edges.equalToSuperview().inset(10)
             $0.height.width.equalTo(20)
         }
     }
+    
+    func setupGestureRecognizers() {
```

Comment:
I don't think we should be doing this, as it's unnecessarily complex and, in essence, unnecessary. 

We can use `didSelectRow...` for opening the `todo form`, while we can create a callback from the `cell` to the `view controller` for the `completed state button`, using a `closure`.